addons/binding/org.eclipse.smarthome.binding.yahooweather/ESH-INF/thing/thing-types.xml

```diff
@@ -27,6 +27,18 @@
                 <description>Specifies the refresh interval in seconds.</description>
                 <default>60</default>
             </parameter>
+
+            <action name="location" type="text">
```

Comment:
> action should be named like the the parameter/-group it's going to provide edit support for

NO - actions and parameters are not intended to be linked in the way you are thinking. You would have a bunch of parameters, and some actions.

You are looking at the test configuration where I just copied the same XML from parameters into action, so yes, there is a lot of redundancy. That's clearly not the intention in real life - this is just the test file. Sorry if this confused matters, but I didn't make the test case real-life, it was just there to test the functionality (sorry).

As I said earlier, actions are there to provide an immediate action - you wouldn't have an action and a parameter that are the same - eg you wouldn't have a reset parameter, and a reset action as that doesn't make sense?

I would expect that a device has a small number of actions where the user needs to trigger something, and a number of parameters where values are set. This is how it is in the zwave binding currently.